src/blocks/scratch3_procedures.js

```diff
@@ -38,6 +38,7 @@ class Scratch3ProcedureBlocks {
 
             const [paramNames, paramIds, paramDefaults] = paramNamesIdsAndDefaults;
 
+            util.initParams();
```

Comment:
Could you add a comment above this line indicating that this initializes the stack frame for the procedure call so that even procedures without parameters get their stack frames initialized.

```suggestion
            // Initialize params for the current stackFrame to {}, even if the procedure does
            // not take any arguments. This is so that `getParam` down the line does not look
            // at earlier stack frames for the values of a given parameter (#1729)
            util.initParams();
```